src/plugins/query_enhancements/public/query_assist/utils/errors.ts

```diff
@@ -12,6 +13,33 @@ export class ProhibitedQueryError extends Error {
   }
 }
 
+interface IAgentError {
+  error: {
```

Comment:
this just got merged today but maybe we could use this: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx#L21

Reply:
i think that one is only used for async query. there are some differences since ml-commons error will have `type` but won't have `status`